modules/utils/include/hephaestus/utils/concepts.h

```diff
@@ -13,6 +13,31 @@ namespace heph {
 template <typename T>
 concept ScalarType = requires(T a) { std::is_scalar_v<T>; };
 
+template <typename T>
+concept EnumType = std::is_enum_v<T>;
+
+template <typename T>
+concept VectorType = requires {
+  typename T::value_type;
+  requires std::same_as<T, std::vector<typename T::value_type>>;
+};
```

Comment:
maybe we can extend to work also with `std::array`?

Reply:
Extended, and added unit tests. PTAL.